keras/utils/file_utils.py

```diff
@@ -437,7 +437,7 @@ def join(path, *paths):
             return gfile.join(path, *paths)
         else:
             _raise_if_no_gfile(path)
-    return os.path.join(path, *paths)
+    return os.path.join(path, *paths).replace("\\", "/")
```

Comment:
What if you use `join` to create a path to refer to a local file on Windows? Does the above change break this use case?

Reply:
Windows can understand forward slashes (/) but os.path.join on Windows will cause backward slashes (\). I believe that a `join` call here causes mixed slashes (i.e. \\ on the last part of the path), which causes the issue.

Replacing all backslashes with / should prevent this. Is there something we can do test cross-platform?

Reply:
I'm not aware of any options for us to test on Windows. GCP has Windows Server VMs, but we've never tried using them.

If the issue is specifically with paths inside our `.keras` files, we could do the string replacement at that level in order to minimize unexpected side effects.

Reply:
Got it, I've moved the string replacement to the saving library, thanks!